### ISSUE TITLE
Pasting a link is really awkward

### ISSUE THREAD

**Issue by jskeet** [benweet/stackedit#1563]
When I create a link in Markdown, I tend to do so manually:

- Write the link text in square brackets
- Write the (
- Paste a link from my URL bar in a different tab
- Write the )

Unfortunately StackEdit decides it wants to handle the link pasting as "create a link with the URL as both the link text and the URL" which means I then need to edit half of it out.

Could it either just not do *anything* special when pasting URLs, or at least detect that this is already the URL part of a link, and disable the special handling in that case?

**Comment by pucherico** [comment]
Instead of handling pasting text I think it would be a nice and productive feature handling pasting and imaging by uploading it transparently and producing a resulting URL.

NOTE: Other editors use imgurg as the external service to upload image (such as HackMD.io)

**Comment by jskeet** [comment]
@pucherico: This isn't about images at all. It's about regular links.

**Comment by pucherico** [comment]
@jskeet I know. Perhaps it'd be better to comment this subject in a new issue.

**Comment by m0o066** [comment]
Have this link issue too. Is there anything I can config in the settings to disable this behavior?

**Comment by ortonomy** [comment]
Came here to talk about this issue, so I'd like to add my two cents:

> Could it either just not do anything special when pasting URLs, or at least detect that this is already the URL part of a link, and disable the special handling in that case?

I *really* like this behaviour (at least the pre-creating of the link format `[]()` !

My suggestion for modified behaviour:

> Keep parsing and pre-constructing the link format, **but leave the link text empty**

E.g. 

- I past the link `https://www.example.com`
- The result in the editor is `[](https://www.example.com)`
- The cursor is left between `[]` the link text

This would be soooooo much more useful

**Comment by jskeet** [comment]
@ortonomy: So what would you expect that to do in my case, where I've already written `[this is the link text](` and I just want to paste the URL? Adding an extra `[]` there would be irritating IMO.

**Comment by ortonomy** [comment]
@jskeet  you already said it, no?

> at least detect that this is already the URL part of a link, and disable the special handling in that case

I still think the current behaviour is useful and needs to be modified along with your suggestion.

**Comment by jskeet** [comment]
@ortonomy: Yes, I suggested one thing - but it wasn't clear that you agreed with that part. You seemed to be *disagreeing* with the overall request.

**Comment by ortonomy** [comment]
@jskeet -- partially:

> Could it either just not do anything special when pasting URLs

I do disagree with this part!  I should have been clearer in my quoting.

**Comment by rafx85** [comment]
Any solution to this? When I'm trying to past link in code the link is converted instead just to past link. Any options to disable it?

**Comment by pucherico** [comment]
@rafx85 My workaround to this is just pasting the link with `<Ctrl>+<Shift>+<V>` instead of `<Ctrl>+<V>`

**Comment by rafx85** [comment]
@pucherico Thank you, will try that.

**Comment by davidegaspar** [comment]
An option to disable this behaviour would be great.
for now the `<Ctrl>+<Shift>+<V>` seems to be the possible workaround
